sdk/spring/azure-spring-cloud-stream-binder-eventhubs/README.md

```diff
@@ -63,59 +65,78 @@ Name | Description | Required | Default
+**_send-timeout_**
 
- Default: `10000`
- 
- #### Event Hub Consumer Properties ####
+Effective only if `sync` is set to true. The amount of time to wait for a response from Event Hub after a send operation, in milliseconds.
+
+Default: `10000`
+
+**_partition-count_**
+
+Optional. Effective only if `partition-key-expression` or `partition-key-extractor-name` is set. The partition size of Event Hub.
+
+Default: 1
+
+**_partition-key-extractor-name_**
+
+Optional. It is the name of a Bean, which implements interface `PartitionKeyExtractorStrategy`.
+`PartitionHandler#extractKey`, the partition handler will first use the `PartitionKeyExtractorStrategy#extractKey` method to apply the message to be sent to obtain the partition key value.
+
+Default: N/A
+
+**_partition-key-expression_**
+
+Optional. It is a SpEL expression that only applies to the message to be sent. When interface `PartitionKeyExtractorStrategy#extractKey` is not implemented, it will be called in the method `PartitionHandler#extractKey`.
+
+Default: N/A
+
+#### Event Hub Consumer Properties ####
+
+It supports the following configurations with the format of `spring.cloud.stream.eventhub.bindings.<channelName>.consumer`.
+
+**_start-position_**
+
+Whether the consumer receives messages from the beginning or end of event hub. if `EARLIEST`, from beginning. If 
+`LATEST`, from end.
+
+Default: `LATEST`
+
+**_checkpoint-mode_**
+
+The mode in which checkpoints are updated.
 
-  It supports the following configurations with the format of `spring.cloud.stream.eventhub.bindings.<channelName>.consumer`.
+`RECORD`, checkpoints occur after each record is successfully processed by user-defined message handler without any exception. If you use `StorageAccount` as checkpoint store, this might become botterneck. 
 
-  **_start-position_**
+`BATCH`, checkpoints occur after each batch of messages successfully processed by user-defined message handler without any exception. `default` mode. You may experience reprocessing at most one batch of messages when message processing fails. Be aware that batch size could be any value.
 
-  Whether the consumer receives messages from the beginning or end of event hub. if `EARLIEST`, from beginning. If 
-  `LATEST`, from end.
+`MANUAL`, checkpoints occur on demand by the user via the `Checkpointer`. You can do checkpoints after the message has been successfully processed. `Message.getHeaders.get(AzureHeaders.CHECKPOINTER)`callback can get you the `Checkpointer` you need. Please be aware all messages in the corresponding Event Hub partition before this message will be considered as successfully processed.
 
-  Default: `LATEST`
+`PARTITION_COUNT`, checkpoints occur after the count of messages defined by `checkpoint_count` successfully processed for each partition. You may experience reprocessing at most `checkpoint_count` of  when message processing fails.
```

Comment:
please help check do we still have this check-point mode?

Reply:
Currently, the check-point-mode is still available.